.github/workflows/web.yaml

```diff
@@ -0,0 +1,48 @@
+# Separate workflow for the wasmer-web crate.
+#
+
+# Automatically cancel previous workflow runs when a new commit is pushed.
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+name: Wasmer Web
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+jobs:
+  web:
+    name: Build and Test (wasmer-web)
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        rust: [nightly]
+    runs-on: ${{ matrix.os }}
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+
+    - name: Install Rust
+      uses: actions/checkout@v3
+      with:
+        toolchain: ${{ matrix.rust }}
+
+    - name: Show Rust version
+      run: |
+        rustc --version
+
+    - name: Check formatting
+      shell: bash
+      run: |
+        cd lib/wasi-web
+        cargo fmt --check
+
+    - name: Check
+      continue-on-error: true
+      env:
+        CARGO_NET_GIT_FETCH_WITH_CLI: "true"
```

Comment:
Optional: do we also want to add this to the makefile and use `make XXX` instead so we have local parity for checks?